ruby/ql/lib/codeql/ruby/typetracking/TypeTrackerSpecific.qll

```diff
@@ -168,24 +209,27 @@ predicate returnStep(Node nodeFrom, Node nodeTo) {
  * called.
  */
 predicate basicStoreStep(Node nodeFrom, Node nodeTo, DataFlow::ContentSet contents) {
-  postUpdateStoreStep(nodeFrom, nodeTo, contents)
+  storeStepIntoSourceNode(nodeFrom, nodeTo, contents)
   or
   exists(
-    SummarizedCallable callable, DataFlowPublic::CallNode call, SummaryComponent input,
-    SummaryComponent output
+    SummarizedCallable callable, DataFlowPublic::CallNode call, SummaryComponentStack input,
+    SummaryComponentStack output
   |
-    hasStoreSummary(callable, contents, input, output) and
+    hasStoreSummary(callable, contents, pragma[only_bind_into](input),
+      pragma[only_bind_into](output)) and
     call.asExpr().getExpr() = callable.getACallSimple() and
-    nodeFrom = evaluateSummaryComponentLocal(call, input) and
-    nodeTo = evaluateSummaryComponentLocal(call, output)
+    nodeFrom = evaluateSummaryComponentStackLocal(callable, call, input) and
+    nodeTo = evaluateSummaryComponentStackLocal(callable, call, output)
```

Comment:
For a store step induced from a flow summary
```ql
input = Argument[0] and
output = Argument[1].Element[?]
```
and a call to the relevant method
```rb
foo(x, y)
```
doesn't this yield a store-step `x -store-> y`, while it should have been `x -store-> [post] y`?

The same for read-store steps.